msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/DiscussionForumTool.java

```diff
@@ -1471,7 +1477,13 @@ public String processActionSaveForumAsDraft()
       setErrorMessage(getResourceBundleString(VALID_FORUM_TITLE_WARN));
       return FORUM_SETTING_REVISE;
     }    
-    saveForumSettings(true);
+    if (selectedForum.getForum().getRestrictPermissionsForGroups() ) {
```

Comment:
This doesn't seem ideal... Ask yourself does it make sense to have a completely separate code path just to create a new forum with these restrictions? Why can't the existing (saveForumSettings) method be adapted?

Lets not go down the road of having so many different methods that essentially do the same thing...

Reply:
But they don't really do the same thing. In fact, I reproduced the path currently followed when creating topics (see https://github.com/sakaiproject/sakai/commit/0641e8c1cdc6c3363b379ee40388d22dc3328d2c).

The main difference is on the saveObjectSettings you only create one topic or forum respectively, whereas on the saveObjectForGroups methods you might create X topics/forums for X different groups, ending each iteration on a call to the saveObjectSettings.

Reply:
I would question whether its being done correctly in topics as well!
I would rather have one saveForumsSettings method that contains the logic than having to know which method to call.
i.e.
``` 
if (some specific restriction) {
  // call some method that saves 
  saveForumsSettingsWithRestriction();
} else {
  // call the normal save forums settings method
  saveForumsSettings();
}
```
or
```
saveForumsSetting() {
  if (some restriction) {
    // do something specific for this restriction
  } else {
    // no restrictions
  }
}
```